### ISSUE TITLE
scaladsl.model.HttpResponse.withStatus return javadsl.model.HttpResponse

### ISSUE THREAD

**Issue by carstenlenz** [akka/akka-http#1623]
Hi.

the following code snippet will not compile:
```scala
import akka.http.scaladsl.model.{HttpResponse, StatusCodes}

val response: HttpResponse = HttpResponse().withStatus(StatusCodes.NotFound)
```
because the `.withStatus()` method return a `javadsl.model.HttpResponse`.

**Expectation**: is should return `scaladsl.model.HttpResponse` and the code should compile.

Opening this after a short discussion in the Gitter channel by request of @johanandren

**Comment by jrudolph** [comment]
Thanks, we seem to be indeed missing overloads that return the more specific Scala types. Weird, that no one noticed so far.

**Comment by carstenlenz** [comment]
Yes - that is why I thought that there is a surely a reason for having it this way. ;)

**Comment by jrudolph** [comment]
Fixed by #1664.
